### PR TITLE
Use github action for kfserving controller tests 

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Use github action for KFServing go controller unit tests 

**Release note**:

```release-note

```

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.14
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
```
Comment (pugangxa) [inline .github/workflows/go.yml:25]: Seems need not do this, but I guess this is for vendor mode right?

Comment (yuzisun) [reply .github/workflows/go.yml:25]: you are right, this is not needed

File: Makefile [hunk 2]
```diff
@@ -184,6 +184,19 @@ docker-build-storageInitializer:
 docker-push-storageInitializer: docker-build-storageInitializer
 	docker push ${KO_DOCKER_REPO}/${STORAGE_INIT_IMG}
 
+kubebuilder:
```
Comment (ellistarn) [inline Makefile:187]: Why do we need kb as a dependency? This is for generating controllers, right?

Comment (yuzisun) [reply Makefile:187]: this is for `make test` which requires local apiserver/etcd which is installed part of kubebuilder

**Pull request merged**